File: CS/MVVMExpenses/Models/CredentialsSource.cs
using System.Collections.Generic;
using DevExpress.Mvvm.POCO;
using MVVMExpenses.Models;

namespace MVVMExpenses.ViewModels
{
    //
    // TODO - your implementation
    static class CredentialsSource {
        static System.Collections.Hashtable credentials;
        static CredentialsSource() {
            credentials = new System.Collections.Hashtable();
            credentials.Add("Guest", GetHash(null));
            credentials.Add("John", GetHash("qwerty"));
            credentials.Add("Administrator", GetHash("admin"));
            credentials.Add("Mary", GetHash("12345"));
        }
        internal static bool Check(string login, string pwd) {
            return object.Equals(credentials[login], GetHash(pwd));
        }
        static object GetHash(string password) {
            return password;
        }
        internal static System.Collections.Generic.IEnumerable<string> GetUserNames() {
            foreach(string item in credentials.Keys)
                yield return item;
        }
    }
}

File: VB/MVVMExpenses/Models/CredentialsSource.vb
Imports System.Collections.Generic
Imports DevExpress.Mvvm.POCO

Namespace MVVMExpenses.Models
    '
    ' TODO - your implementation
    Friend NotInheritable Class CredentialsSource

        Private Sub New()
        End Sub

        Private Shared credentials As System.Collections.Hashtable
        Shared Sub New()
            credentials = New System.Collections.Hashtable()
            credentials.Add("Guest", GetHash(Nothing))
            credentials.Add("John", GetHash("qwerty"))
            credentials.Add("Administrator", GetHash("admin"))
            credentials.Add("Mary", GetHash("12345"))
        End Sub
        Friend Shared Function Check(ByVal login As String, ByVal pwd As String) As Boolean
            Return Object.Equals(credentials(login), GetHash(pwd))
        End Function
        Private Shared Function GetHash(ByVal password As String) As Object
            Return password
        End Function
        Friend Shared Iterator Function GetUserNames() As System.Collections.Generic.IEnumerable(Of String)
            For Each item As String In credentials.Keys
                Yield item
            Next item
        End Function
    End Class
End Namespace
